script.js

```diff
@@ -2,27 +2,85 @@ const form = document.querySelector('form');
 const input = document.querySelector('#txtTaskName');
 const btnDeleteAll = document.querySelector('#btnDeleteAll');
 const taskList = document.querySelector('#task-list');
+let items;
 eventListeners();
+loadItems();
 
 function eventListeners() {
     form.addEventListener('submit', addNewItem);
-}
-
-function addNewItem(e) {
-    if (input.value === '') {
-        alert('Add New Item Please')
+    taskList.addEventListener('click', deleteItem)
+    btnDeleteAll.addEventListener('click', deleteAllItem)
+};
+function loadItems() {
+    items=getItemsFromLS();
+    items.forEach(function(item){
+        createItem(item);
+    })
+};
+function getItemsFromLS(){
+    if(localStorage.getItem('items')===null){
+        items=[];
+    }else{
+        items=JSON.parse(localStorage.getItem('items'))
     }
+    return items;
+};
+function setItemToLS(text){
+    items=getItemsFromLS();
+    items.push(text);
+    localStorage.setItem('items',JSON.stringify(items));
+};
+function deleteItemFromLS(text){
+    items=getItemsFromLS();
+    items.forEach(function(item,index){
+        if(item===text){
+            items.splice(index,1);
+        }
+    });
+    localStorage.setItem('items', JSON.stringify(items));
+};
+function createItem(text) {
     const li = document.createElement('li');
     li.className = 'list-group-item list-group-item-secondary';
-    li.appendChild(document.createTextNode(input.value));
+    li.appendChild(document.createTextNode(text));
     const a = document.createElement('a');
     a.classList = ('delete-item float-right');
     a.setAttribute('href', '#');
     a.innerHTML = '<i class="fas fa-times"></i>';
     li.appendChild(a);
     taskList.appendChild(li);
     console.log(li);
+};
+function addNewItem(e) {
+    if (input.value === '') {
+        alert('Add New Item Please')
+    }
+    createItem(input.value);
+    setItemToLS(input.value);
     input.value = "";
-    // input.value("");
     e.preventDefault();
-}
\ No newline at end of file
+};
+function deleteItem(e) {
+    if (e.target.className === "fas fa-times") {
+            if (confirm('Are You Sure ?')) {
+            e.target.parentElement.parentElement.remove();
+            deleteItemFromLS(e.target.parentElement.parentElement.textContent);
+        }
+    }
+    e.preventDefault();
+};
+function deleteAllItem(e) {
+    if (confirm('Are You Sure ?')) {
+        // taskList.innerHTML = '';
+        // taskList.childNodes.forEach(function (item) {
+        //     if (item.nodeType === 1) {
+        //         item.remove();
+        //     }
+        // });
+        while (taskList.firstChild) {
+            taskList.removeChild(taskList.firstChild);
+        }
+        localStorage.clear();
+    }
+    e.preventDefault();
+};
```

Comment:
### HTML dosyasi icin: 

- satir ve sonrasindaki Todo itemlari sil. Onlar localstorage dan gelebilir. Yada hic olmayabilir. 
- bootstrap siniflari ile yapilan tasarimi kendi css tasarimina cevirmeyi deneyebilirsin. Güzel bir alistirma olur.
- Todo itemleri icinde neden a tagi var? Bunlar link degil ki. Bir yere gitmiyorlar tiklayinca.
- Form kullanmak yerine bir input ve button ile de bu isi halledebilirsin. Formu kaldir ve öyle yap. Güzel bir alistirma olur ve kullandigin seyler iyice oturur. 
- title My Task List degil de Todo App olmali. Uygulama adi my task list degil todo app. 
- Nesne yönelimli programlara alistirmasi: Bir item sinifi olsa ve sen bu siniftan objeler üretsen, todo item olarak?

Reply:
1-Anladım sanırım.
2- Bunu yapabileceğimi sanmıyorum.
3- Şayet bir yere gönderecek olsaydım nasıl yapılacağını öreneyim diye bıraktım onu.
4-Bunu deneyecem.
5-Düzeltilecek.
6-Bahsettiğin şekilde yapmıştık ama bilgileri local Storage dan çekelim öyle yapalım muhabbeti olunca o kısmı tekrar sildim.